catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/RemoveAllCommandTest.java

```diff
@@ -72,35 +108,77 @@ public void testBadBatchSize() throws Exception {
     }
 
     /**
+     * Tests that if a current batch to be removed is less than the batch size,
+     * but items still remain to be deleted, command will not terminate prematurely.
+     * <p>
+     * Cumulative results by each batch:
+     * First:       53
+     * Second:      106
+     * Third:       159
+     * Fourth:      212 (stops at 201)
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testDeleteFewerThanBatchSizeExpectMultipleCatalogCalls() throws Exception {
+
+        resultsPerCatalogQuery = 53;
+        batchSize = 101;
+        numCatalogCalls = 6;
```

Comment:
✏️ I think you need more comments to explain this test. I fundamentally don't understand it. The `delete` method is invoked six times? Why is it six and not four? Why do you return the `metacardMockList` three times followed by an empty list? Shouldn't it be three times plus another list of 42?

This test is intimately aware of and tied to the internals - which may change in the future - of the way the catalog framework works. It's fragile and not clear.